options/options.go

```diff
@@ -1736,29 +1736,19 @@ func (interp *interpreter) messageLiteralValue(
 				}
 			} else {
 				ffld = fmd.Fields().ByName(protoreflect.Name(fieldNode.Name.Value()))
-				// Groups are indicated in the text format by the group name (which is
-				// camel-case), NOT the field name (which is lower-case).
-				// ...but only regular fields, not extensions that are groups...
-				if ffld != nil && ffld.Kind() == protoreflect.GroupKind &&
-					string(ffld.Name()) == strings.ToLower(string(ffld.Message().Name())) &&
-					ffld.Message().Name() != protoreflect.Name(fieldNode.Name.Value()) {
-					// This is kind of silly to fail here, but this mimics protoc behavior.
-					// We only fail when this really looks like a group since we need to be
-					// able to use the field name for fields in editions files that use the
-					// delimited message encoding and don't use proto2 group naming.
-					return protoreflect.Value{}, sourceinfo.OptionSourceInfo{},
-						reporter.Errorf(interp.nodeInfo(fieldNode.Name), "%vfield %s not found (did you mean the group named %s?)", mc, fieldNode.Name.Value(), ffld.Message().Name())
-				}
 				if ffld == nil {
 					err = protoregistry.NotFound
-					// could be a group name
-					for i := 0; i < fmd.Fields().Len(); i++ {
-						fd := fmd.Fields().Get(i)
-						if fd.Kind() == protoreflect.GroupKind && fd.Message().Name() == protoreflect.Name(fieldNode.Name.Value()) {
-							// found it!
-							ffld = fd
+					// It could be a group, where the field-name is the lower-case
+					// form of the group type name.
+					ffld = fmd.Fields().ByName(protoreflect.Name(strings.ToLower(fieldNode.Name.Value())))
+					if ffld != nil {
+						if ffld.Kind() != protoreflect.GroupKind ||
```

Comment:
Do you need to check features for `DELIMITED`? It may be worth commenting these checks similar to protoc here: https://github.com/protocolbuffers/protobuf/commit/29c69ff00b58b60e67fcf40fd810009bd39b86c6#diff-2228551d02c6661809ca7103db9512eef4c2d01f35556d42316543d92a89edefR1479-R1491

Reply:
The check of `ffld.Kind()` already incorporates that: https://github.com/bufbuild/protocompile/blame/main/linker/descriptors.go#L1064-L1074

Reply:
I pushed a commit that tweaks this a little bit and adds some comments. Better?

BTW, the reason we are not explicitly checking if the message and field are in the same file (which is check that is present in that `protoc` link) is because that is not necessary here. At this point in the logic, we know it's _not_ an extension. It is only possible for the message and field to be in the same scope but in. _different_ files if the group is an extension (where the "same scope" just means in the same package). For non-extension fields, the scope check is enough and guarantees that the message type is nested inside the same type that defines the field.

Reply:
Thanks for the detailed comments.